rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderBridgeTxsIndexTests.java

```diff
@@ -0,0 +1,378 @@
+        );
+        // reset calls counter
+        Mockito.reset(repository);
+
+        // Let's set the sigHash and then check if the hasBtcTxSigHash returns true
+        provider.setBridgeBtcTxSigHash(sigHash);
+        verify(repository, times(1)).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+        // reset calls counter
+        Mockito.reset(repository);
+
+        // The entry should be found in the pending to save set, not in the index
+        boolean shouldFoundSigHash = provider.hasBridgeBtcTxSigHash(sigHash);
+        Assertions.assertTrue(shouldFoundSigHash);
+        verify(repository, never()).getStorageBytes(
+            any(),
+            any()
+        );
+        Mockito.reset(repository);
+
+        // Let's save into the repository the pending to save sigHash
+        provider.save();
+        verify(repository, times(1)).addStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString()),
+            new byte[]{(byte)1}
+        );
+        verify(repository, never()).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+        Mockito.reset(repository);
+
+        // Check if sigHash exists when there are entries in the index but not pending to save
+        boolean shouldFoundAgainSigHash = provider.hasBridgeBtcTxSigHash(sigHash);
+        Assertions.assertTrue(shouldFoundAgainSigHash);
+        verify(repository, never()).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            BRIDGE_BTC_TX_SIG_HASHES_KEY.getCompoundKey("-", sigHash.toString())
+        );
+
+        // Create a new bridge storage provider instance to pretend the node is processing a brand-new transaction
+        provider = new BridgeStorageProvider(
+            repository,
+            PrecompiledContracts.BRIDGE_ADDR,
+            bridgeConstantsRegtest,
+            activations
+        );
+
+        // Let's check nothing is persisted when calling save and there are not pending entries to save
+        provider.save();
+        verify(repository, never()).addStorageBytes(
+            any(),
+            any(),
+            any()
+        );
+        Mockito.reset(repository);
+
+        // Try to set a new sigHash that no exists in memory not in the index.
+        Sha256Hash newSigHash = PegTestUtils.createHash(7);
+        provider.setBridgeBtcTxSigHash(newSigHash);
+
+        // Verify the new sigHash is not persisted into the index when save has not been called.
+        verify(repository, never()).addStorageBytes(
+            any(),
+            any(),
+            any()
+        );
+
+        // Now let's persist pending to save entries
+        provider.save();
+        verify(repository, times(1)).addStorageBytes(
```

Comment:
Here I think it could go like:

```java
verify(repository, times(1)).addStorageBytes(
            any(RskAddress.class),
            any(DataWord.class),
            any(byte[].class)
        );
```

What do you think?

Reply:
Done